src/colors.js

```diff
@@ -0,0 +1,57 @@
+import {color} from 'chart.js/helpers';
+import {getGradientData} from './helpers';
+
+const rgbs = (c) => Math.round(c._rgb.a * 255) << 24 | c._rgb.r << 16 | c._rgb.g << 8 | c._rgb.b;
+// IEC 61966-2-1:1999
+const toRGBs = (l) => l <= 0.0031308 ? l * 12.92 : Math.pow(l, 1.0 / 2.4) * 1.055 - 0.055;
+// IEC 61966-2-1:1999
+const fromRGBs = (srgb) => srgb <= 0.04045 ? srgb / 12.92 : Math.pow((srgb + 0.055) / 1.055, 2.4);
+
+function interpolate(percent, startColor, endColor) {
+  const startRGBs = rgbs(startColor.color);
+  const startR = fromRGBs(((startRGBs >> 16) & 0xff) / 255);
+  const startG = fromRGBs(((startRGBs >> 8) & 0xff) / 255);
+  const startB = fromRGBs((startRGBs & 0xff) / 255);
```

Comment:
No need to convert to int and back?

```suggestion
  const start = startColor.color.rgb;
  const startR = fromRGBs(start.r / 255);
  const startG = fromRGBs(start.g / 255);
  const startB = fromRGBs(start.b / 255);
```